### ISSUE TITLE
09-k-means-housing.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/Eder_Zelaya-Machine-Learnig#5]
Hola @Eder, por aquí las issues!

## LO QUE HAS HECHO BIEN

- **División train/test ANTES del clustering**: Perfecto, es exactamente como pide el ejercicio. Entrenas K-Means solo con train y luego predices sobre test
- **6 clusters**: Correcto, usas `n_clusters=6` como pide el enunciado
- **Predicción sobre test**: Bien implementado con `model_1.predict(X_test)`
- **Visualización comparativa train vs test**: Excelente idea usar `alpha=0.1` para train y `marker="+"` para test, permite ver ambos conjuntos superpuestos
- **Múltiples visualizaciones**: Muy bien mostrar las 3 combinaciones de variables (Lat-Long, Lat-MedInc, Long-MedInc)
- **Observaciones detalladas por cluster**: Buenas interpretaciones de qué representa cada cluster geográfica y económicamente
- **Modelo supervisado**: Correcto uso de DecisionTreeClassifier con los clusters como target
- **Imports organizados**: Aunque hay algunos que no usas, están bien agrupados

### 1. Falta escalar los datos (IMPORTANTE)
K-Means es sensible a la escala de las variables. `MedInc` tiene un rango diferente a `Latitude` y `Longitude`. Deberías escalar:

```python
from sklearn.preprocessing import StandardScaler

scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)  # Solo transform, NO fit_transform

model_1.fit(X_train_scaled)
```

### 2. No eliminas duplicados correctamente
Solo verificas si hay duplicados pero no los eliminas:

```python
# Tu código solo muestra True/False
df.duplicated()

# Falta eliminarlos
df.drop_duplicates(inplace=True)
```

### 3. No guardas los modelos (Paso 5)
El ejercicio pide explícitamente:
> "Almacena **ambos modelos** en la carpeta correspondiente"

Tienes `joblib` importado pero no lo usas:

```python
# Guardar K-Means
joblib.dump(model_1, '../models/kmeans_housing.pkl')

# Guardar DecisionTree
joblib.dump(model_2, '../models/decision_tree_housing.pkl')

# También el scaler si lo usas
joblib.dump(scaler, '../models/scaler_housing.pkl')
```

### 4. Faltan métricas completas del modelo supervisado
Solo muestras accuracy. Faltan confusion_matrix y classification_report:

```python
print(f"Accuracy: {accuracy_score(y_test_sup, y_pred)}")
print("\nConfusion Matrix:")
print(confusion_matrix(y_test_sup, y_pred))
print("\nClassification Report:")
print(classification_report(y_test_sup, y_pred))
```

### 5. Error en el plot del árbol: usas columnas incorrectas
```python
# Tu código
tree.plot_tree(model_2, feature_names = list(X_train.columns), ...)

# X_train tiene 4 columnas: MedInc, Latitude, Longitude, cluster
# Pero el modelo se entrena con X_train_sup que tiene 3: MedInc, Latitude, Longitude

# Lo correcto:
tree.plot_tree(model_2, feature_names = list(X_train_sup.columns), ...)
```

### 6. Imports innecesarios
Tienes imports que no usas en este ejercicio:
- `load_iris`, `fetch_20newsgroups`
- `TfidfVectorizer`, `CountVectorizer`
- `RandomizedSearchCV`, `randint`, `uniform`
- `RandomForestClassifier`, `KNeighborsClassifier`

No es un error grave, pero limpia el código.

### 7. Falta descripción/conclusión del modelo supervisado
Al final deberías describir los resultados del DecisionTree:
- ¿Qué accuracy obtuviste?
- ¿Generaliza bien el modelo?
- ¿Por qué elegiste DecisionTree y no otro modelo?

---

Buen trabajo Eder, el flujo del ejercicio está bien entendido (split → K-Means en train → predict en test → visualizar → modelo supervisado). Los puntos principales a mejorar son: escalar los datos antes de K-Means, eliminar duplicados correctamente, guardar los modelos, y completar las métricas. El notebook está bien estructurado y las visualizaciones comparativas están muy bien hechas!
